packages/widget-v2/src/pages/SwapPage/useSetMaxAmount.ts

```diff
@@ -0,0 +1,75 @@
+
+import { convertHumanReadableAmountToCryptoAmount, convertTokenAmountToHumanReadableAmount } from "@/utils/crypto";
+import { useGetAssetDetails } from "@/hooks/useGetAssetDetails";
+import { sourceAssetAmountAtom, sourceAssetAtom } from "@/state/swapPage";
+import { useAtom, useSetAtom } from "jotai";
+import { skipChainsAtom } from "@/state/skipClient";
+import { useSourceBalance } from "./useSourceBalance";
+import { BigNumber } from "bignumber.js";
+
+
+const ETH_GAS_FEE = 0.01;
+const COSMOS_GAS_FEE = 2_000_000;
+
+export const useGasFeeTokenAmount = () => {
+  const [sourceAsset] = useAtom(sourceAssetAtom);
+  const [{ data: chains }] = useAtom(skipChainsAtom);
+
+  const sourceDetails = useGetAssetDetails({
+    assetDenom: sourceAsset?.denom,
+    amount: sourceAsset?.amount,
+    chainId: sourceAsset?.chainID,
+  });
+
+  const feeAsset = chains?.find(chain => chain.chainID === sourceAsset?.chainID)?.feeAssets?.[0];
+
+  const chainType = sourceDetails?.chain?.chainType;
+
+  switch (chainType) {
+    case "evm":
+      return Number(convertHumanReadableAmountToCryptoAmount(ETH_GAS_FEE));
```

Comment:
you need to pass decimals from source details, we are not sure every evm native asset is 18 decimals

Reply:
@codingki gotcha so the fee for evm is always in the source asset